src/block_cache.cpp

```diff
@@ -460,7 +460,7 @@ void block_cache::cache_hit(cached_piece_entry* p, void* requester, bool volatil
 	// move into L2 (frequently used)
 	m_lru[p->cache_state].erase(p);
 	m_lru[target_queue].push_back(p);
-	p->cache_state = target_queue;
+	p->cache_state = std::uint16_t(target_queue);
```

Comment:
it seems ``target_queue`` should have the right type here.. maybe it should even be an enum class